kolibri-electron/src/index.js

```diff
@@ -10,14 +14,124 @@ if (require('electron-squirrel-startup')) { // eslint-disable-line global-requir
 }
 
 const KOLIBRI = 'http://localhost:5000';
-const pingTimeout = 20;
+let pingTimeout = 20;
+let mainWindow = null;
 let loadRetries = 0;
 let timeSpent = 0;
 let maxRetries = 3;
 
 let django = null;
 
-const waitForKolibriUp = (mainWindow) => {
+async function getEndlessKeyDrive() {
+  const drives = await drivelist.list();
+
+  // Look for the endless key drive
+  for (const d of drives.map((d) => d.mountpoints[0].path)) {
+    console.log(`Looking for endless Key on ${d}`);
+    const p = path.join(d, 'KOLIBRI_DATA');
+    try {
+      await fsPromises.access(p);
+      return d;
+    } catch (err) {
+      console.log(`${p} doesn't exists, trying next`);
+    }
+  };
+
+  return null;
+}
+
+async function getLoadingScreen() {
+  const defaultLoading = path.join(__dirname, 'Kolibri', 'assets', '_load.html');
+
+  // Try to use the custom loading screen from kolibri-explore-plugin
+  const drive = await getEndlessKeyDrive();
+  if (!drive) {
+    return defaultLoading;
+  }
+
+  const loading = path.join(
+    drive,
+    'KOLIBRI_DATA',
+    'extensions',
+    'kolibri_explore_plugin',
+    'loadingScreen',
+    'index.html',
+  );
+
+  try {
+    await fsPromises.access(loading);
+    return loading;
+  } catch (err) {
+    console.log(`Loading screen not found ${loading}`);
+  }
+
+  return defaultLoading;
+}
+
+async function getPluginVersion() {
+  const drive = await getEndlessKeyDrive();
+  if (!drive) {
+    console.error('Endless Key not found');
+    return '0.0.0';
```

Comment:
Let's split this into a constant, `NULL_PLUGIN_VERSION`, since we're using it in a few places. I think we could safely set it to '0', but I also won't mind the long version format if it isn't being repeated :)

Reply:
done